lib/nerves_hub_web/live/devices/show.ex

```diff
@@ -362,4 +364,11 @@ defmodule NervesHubWeb.Live.Devices.Show do
       device.connecting_code
     end
   end
+
+  defp format_alarm(alarm), do: String.trim_leading(alarm, "Elixir.")
+
+  defp has_description?(""), do: false
+  defp has_description?(description) when is_binary(description), do: true
+  # To cover for cases where alarm description is provided as an invalid type.
+  defp has_description?(_), do: false
```

Comment:
```suggestion
  defp has_description?(description) do
    is_binary(description) and byte_size(description) > 0
  end
```
I also think we could reduce overhead here with an all-encompassing check rather than patterns to clearly show what is going on